packages/evm/src/fungible.ts

```diff
@@ -95,18 +98,31 @@ export async function createEvmFungibleAssetTransfer(
 class EvmFungibleAssetTransfer extends EvmTransfer {
   protected destinationAddress: string = '';
   protected securityModel: SecurityModel;
-  protected _amount: bigint;
+  protected _adjustedAmount: bigint = BigInt(0);
+  private specifiedAmount: bigint;
+
+  constructor(transfer: EvmFungibleTransferRequest, config: Config) {
+    super(transfer, config);
+    this.specifiedAmount = transfer.amount;
 
-  get amount(): bigint {
-    return this._amount;
+    if (isValidAddressForNetwork(transfer.destinationAddress, this.destination.type))
+      this.destinationAddress = transfer.destinationAddress;
+    this.securityModel = transfer.securityModel ?? SecurityModel.MPC;
+  }
+
+  /**
+   * Returns amount to be transferred considering the fee
+   */
+  get adjustedAmount(): bigint {
```

Comment:
instead of having a `getAdjustAmount` getter can we have a simple `amount` getter, no need to provide context to end user I would say